static/js/src/openstack/react/components/CostCalculatorForm.jsx

```diff
@@ -0,0 +1,223 @@
+import React, { useState } from "react";
+import {
+  Row,
+  Col,
+  Slider,
+  Input,
+  Form,
+  Tooltip,
+  Icon,
+} from "@canonical/react-components";
+
+import CostCalculations from "./CostCalculations";
+
+const CostCalculatorForm = () => {
+  const [formState, setFormState] = useState({
+    instances: { value: 1000, error: "" },
+    vcpus: { value: 2, error: "" },
+    ephemeralStorage: { value: 8, error: "" },
+    ram: { value: 8, error: "" },
+    persistentStorage: { value: 80, error: "" },
+    supportLevel: "fully-managed",
+  });
+
+  const onChangeHandler = (e) => {
+    const value =
+      e.target.type === "checkbox" ? e.target.checked : e.target.value;
+
+    const error =
+      parseInt(e.target.value) < parseInt(e.target.min) ||
+      parseInt(e.target.value) > parseInt(e.target.max) ||
+      e.target.value === ""
+        ? `Please enter a value between ${e.target.min} and ${e.target.max}`
+        : false;
+
+    e.target.name !== "supportLevel"
+      ? setFormState({
+          ...formState,
+          [e.target.name]: { value: parseInt(value), error: error },
+        })
+      : setFormState({ ...formState, supportLevel: value });
+  };
+
+  return (
+    <>
+      <Form>
+        <Row>
+          <Col size="6">
+            <h3 class="p-heading--4">
+              Number of instances
+              <Tooltip
+                position="top-right"
+                message="A number of virtual machines that will run in the cloud."
+              >
+                <Icon name="help" className="tco-tooltip" />
+              </Tooltip>
+            </h3>
+            <Slider
+              max={10000}
+              mdxType="Slider"
+              min={1}
+              name="instances"
+              id="instances"
+              onChange={onChangeHandler}
+              showInput
+              value={formState.instances.value}
+              error={formState.instances.error}
+              ariaLabel="Number of instances"
+              required
+            />
+            <h3 class="p-heading--4" style={{ marginTop: "2rem" }}>
+              Instance type
+              <Tooltip
+                position="top-right"
+                message=" Average virtual machine resources configuration."
+              >
+                <Icon name="help" className="tco-tooltip" />
+              </Tooltip>
+            </h3>
+            <Row>
+              <Col size="3">
+                <Row>
+                  <Col size="3">
```

Comment:
This is quite confusing and it's abusing the wrapping natural of a misconfigured set of cols in a grid. I would suggest, moving all these elements to the top-level `Col` and adding max-width style to the inputs.